### PR TITLE
Honor maven localRepository setting in settings.xml files.

### PR DESCRIPTION
**Pull request by xkr47**
Scan ~/.m2/settings.xml and ${M2_HOME}/conf/settings.xml for a localRepository setting and make maven ant tasks use it if found.

### CODE REVIEW

**Comment by xkr47** [comment]
The maven ant task "mvn:mvn" does not read the settings.xml files at all by default. By the looks of the source code of that particular task it was written in a few hours and these things were not thought through. Some of the other maven ant tasks do seem to have some support for actually reading the settings.xml files. Beats me..

**Comment by quintesse** [comment]
I read yesterday that the Maven Ant task is deprecated (in 2011 already) and we should actually be using the Aether Ant task. But I couldn't make much sense of its documentation to be honest :/

File: build.xml [hunk 1]
```diff
@@ -21,11 +25,25 @@
...
+    <available file="${m2-user-settings}" property="m2-user-settings-exists"/>
+    <available file="${m2-global-settings}" property="m2-global-settings-exists"/>
+    <!-- since properties are read-only, the first assignment remains effective (if any) -->
+    <xmltask source="${m2-user-settings}" if:set="m2-user-settings-exists">
+        <copy path="/*[local-name()='settings']/*[local-name()='localRepository']/text()" property="localRepository"/>
+    </xmltask>
+    <xmltask source="${m2-global-settings}" if:set="m2-global-settings-exists">
+        <copy path="/*[local-name()='settings']/*[local-name()='localRepository']/text()" property="localRepository"/>
+    </xmltask>
+    <echo if:set="localRepository">Using localRepository ${localRepository}</echo>
```
Comment (lucaswerkmeister) [inline build.xml]: Is this `<echo>` necessary?

Comment (xkr47) [inline build.xml]: @lucaswerkmeister No, it's informative :)

**Comment by quintesse** [comment]
@xkr47 I just removed the use of `<mvn:mvn>` from the IDE build file in favor of simply using `<exec>`, could you see if that works for you?
And in case it doesn't work and something needs to be changed please send a new PR and promise to merge it ASAP :)

**Pull request closed without merging**